subsystem/cluster_test.go

```diff
@@ -176,41 +177,74 @@ var _ = Describe("Cluster tests", func() {
 	})
 })
 
+func isClusterInState(ctx context.Context, clusterID strfmt.UUID, state, stateInfo string) (bool, string) {
+	rep, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
+	Expect(err).NotTo(HaveOccurred())
+	c := rep.GetPayload()
+	if swag.StringValue(c.Status) == state {
+		return stateInfo == IgnoreStateInfo || swag.StringValue(c.StatusInfo) == stateInfo, swag.StringValue(c.Status)
+	}
+
+	return false, swag.StringValue(c.Status)
+}
+
 func waitForClusterState(ctx context.Context, clusterID strfmt.UUID, state string, timeout time.Duration, stateInfo string) {
 	log.Infof("Waiting for cluster %s status %s", clusterID, state)
-	for start := time.Now(); time.Since(start) < timeout; {
-		rep, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
-		Expect(err).NotTo(HaveOccurred())
-		c := rep.GetPayload()
-		if swag.StringValue(c.Status) == state {
-			break
+	var lastState string = ""
+	var success bool
+
+	for start, successInRow := time.Now(), 0; time.Since(start) < timeout; {
+		success, lastState = isClusterInState(ctx, clusterID, state, stateInfo)
+
+		if success {
+			successInRow++
+		} else {
+			successInRow = 0
 		}
+
+		// Wait for cluster state to be consistent
+		if successInRow >= minSuccessesInRow {
+			return
+		}
+
 		time.Sleep(time.Second)
 	}
-	rep, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
+
+	Fail(fmt.Sprintf("Cluster %s wasn't in state %s for %d times in a row. Last state was %s",
```

Comment:
Please change it to Expect the diff will be clearer and no one will need to read the message.